enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,580 @@
+
+For the first deliverable, a local deployment of the backing service will
+include new controllers that will operate to achieve the desired state as
+declared in CRDs. Thus the service will expose both REST and Kubernetes-native
+APIs, and both API layers will call the same set of internal methods. The local
+service will include a database; sqlite if possible, or else postgresql.
+
+    --------------------------
+    |  REST API  |  k8s API  |
+    --------------------------
+    | Assisted Service logic |
+    --------------------------
+        |               |
+        v               v
+      SQL DB        File System
+                      storage
+
+For hub clusters that are used for multi-cluster creation and management,
+persistent storage will be a requirement for the database and file system
+storage.
+
+Agents that are running on hardware and waiting for installation-related
+instructions will continue to communicate with the assisted service using the
+existing REST API. In the future that may transition to using a CRD directly.
+In the meantime, status will be propagated to the appropriate k8s resource when
+an agent reports relevant information.
+
+**InstallEnv**
+This new resource, part of the assisted intaller's new operator, represents an
+environment in which a group of hosts share settings related to networking,
+local services, etc. This resource is used to create a discovery image that
+should be used for booting hosts. In the REST API this corresponds to the
+"image" resource that is embedded in the "cluster" resource, but for
+kubernetes-native it is more natural for it to be separate.
+
+The discovery ISO can be downloaded from a URL that is available in the
+resource's Status.
+
+The details of this resource definition are being discussed [in a
+pull request](https://github.com/openshift/assisted-service/pull/969).
+
+**ClusterDeployment**
+Hive's ClusterDeployment CRD will be extended to include all cluster details
+that the agent-based install needs. There will not be a new cluster resource.
+The contents of this API correlate to the "cluster" resource in the assisted
+installer's current REST API.
+
+The details of this are being discussed on a Hive
+[pull request](https://github.com/openshift/hive/pull/1247).
+
+
+**Agent**
+Agent is a new resource, part of the assisted installer's new operator, that
+represents a host that is destined to become part of an OpenShift cluster, and
+is running an agent that is able to run inspection and installation tasks. It
+correlates to the "host" resource in the Assisted Installer's REST API.
+
+The details of this API are being discussed in a [pull
+request](https://github.com/openshift/assisted-service/pull/861) that
+implements the CRD.
+
+
+#### REST API Access
+
+Some REST APIs need to be exposed in addition to the Kubernetes-native APIs
+described below.
+* Download ISO and PXE artifacts: These files must be available for download
```

Comment:
(Minor point - would be helpful to explicitly say discovery ISO here)

Where do PXE artifacts come into this? Are these published in InstallEnv in addition to the discovery ISO?

Reply:
I would say "rootfs" instead of "PXE artifacts".  In the minimal ISO scenario, the ISO doesn't contain the rootfs itself but rather a URL to download it from.  Other PXE artifacts aren't relevant to these flows.